sdks/python/apache_beam/testing/util.py

```diff
@@ -166,23 +166,27 @@ def _equal(actual):
       sorted_actual = sorted(actual)
       if sorted_expected != sorted_actual:
         raise BeamAssertException(
-            'Failed assert: %r == %r' % (sorted_expected, sorted_actual))
+            'Failed assert: %r == %r' % (sorted_actual, sorted_expected))
     # Slower method, used in two cases:
     # 1) If sorted expected != actual, use this method to verify the inequality.
     #    This ensures we don't raise any false negatives for types that don't
     #    have a deterministic sort order.
     # 2) As a fallback if we encounter a TypeError in python 3. this method
     #    works on collections that have different types.
     except (BeamAssertException, TypeError):
+      left_missing = []
       for element in actual:
         try:
           expected_list.remove(element)
         except ValueError:
-          raise BeamAssertException(
-              'Failed assert: %r == %r' % (expected, actual))
-      if expected_list:
-        raise BeamAssertException(
-            'Failed assert: %r == %r' % (expected, actual))
+          left_missing.append(element)
+      if left_missing or expected_list:
+        msg = 'Failed assert: %r == %r' % (actual, expected)
+        if left_missing:
+          msg = msg + ', left side missing elements %r' % left_missing
```

Comment:
Since currently LHS is actual, RHS is expected, I think the error should say "right side is missing..." or "unexpected elements: ...", and `left_missing` should be renamed.